# Iris Dataset Analysis
This notebook analyses the Iris dataset using Python. The aim is to explore the dataset, summarise variables, and identify patterns between different species of iris flowers.

In [1]:
# Import Libraries
import pandas as pd
import matplotlib.pyplot as plt

### Importing Libraries
The pandas library is used for manipulating and analysis, while matplotlib is used for creating visualisations such as graphs and charts.

In [2]:
# Load the dataset
columns = ["sepal_length", "sepal_width", "petal_length", "petal_width", "species"]

df = pd.read_csv("data/iris.csv", names=columns)
df.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,Iris-setosa
1,4.9,3.0,1.4,0.2,Iris-setosa
2,4.7,3.2,1.3,0.2,Iris-setosa
3,4.6,3.1,1.5,0.2,Iris-setosa
4,5.0,3.6,1.4,0.2,Iris-setosa


### Loading the Dataset
The dataset is loaded using pandas. Since the dataset does not include column headers, column names are assigned manually. The first few rows are displayed to confirm that the data has been loaded correctly.

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


## Dataset Structure
The dataset contains 150 entries with five columns: four numerical features and one categorical variable representing the species. This helps to understand the structure and data types.

In [4]:
# Summary statistics
summary = df.describe()
summary

,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.054000,3.758667,1.198667
std,0.828066,0.433594,1.764420,0.763161
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


### Summary Statistics
The summary statistics provide key information such as the mean, minimum, maximum, and standard deviation for each numerical variable. This helps understand the distribution and spread of data.

In [5]:
with open("summary.txt", "w") as f:
    f.write(summary.to_string())

### Save summary
The summary statistic is saved to a text file called 'summary.txt'. This allows the results to be stored and accessed outside of the notebook.